tests/integration_tests/test_lm/test_mpirun.py

```diff
@@ -16,27 +16,30 @@ class TestTask(TestCase):
     # --------------------------------------------------------------------------
     #
     def setUp(self) -> dict:
```

Comment:
I'd like to propose here to start to fix method `setUp` within `TestCase`-based classes. This method shouldn't return anything, rather it prepares the instance of the class for the coming test ([link](https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.setUp)) and it is called before each test (thus no need to call it manually)
```python
    def setUp(self):
        path = os.path.dirname(__file__) + '/../test_config/resources.json'
        resources = ru.read_json(path)
        hostname = socket.gethostname()

        for host in resources.keys():
            if host in hostname:
                self.host = host
                self.resource = resources[host]
                break
```
but if that information will be used for every test (within the test class), then corresponding class setup method `setUpClass` ([link](https://docs.python.org/3.7/library/unittest.html#unittest.TestCase.setUpClass)) should be used, which is called once before all tests start.
```python
    @classmethod
    def setUpClass(self):
        path = os.path.dirname(__file__) + '/../test_config/resources.json'
        resources = ru.read_json(path)
        hostname = socket.gethostname()

        for host in resources.keys():
            if host in hostname:
                cls.host = host
                cls.resource = resources[host]
                break
```
(*) same applied for `tearDown` and `tearDownClass` methods (which could be used for cleaning after each test and after all tests accordingly)

Reply:
Let me check this out and get back at you.